include/solvers/heat_transfer.h

```diff
@@ -129,6 +130,12 @@ class HeatTransfer : public AuxiliaryPhysics<dim, GlobalVectorType>
     // outputs
     if (simulation_parameters.timer.type == Parameters::Timer::Type::none)
       this->computing_timer.disable_output();
+
+    if (simulation_parameters.post_processing.calculate_average_temp_and_hf)
+      {
+        average_temperature =
+          std::make_shared<AverageScalarInTime<dim>>(this->dof_handler);
```

Comment:
since we call the other one AverageVelocity I would maybe call it AverageScalar instead.